### DISCUSSION TITLE
issue with list of strings

### DISCUSSION THREAD

**Discussion by chrisdrackett** [prisma/prisma#2677] [Q&A] [answered]
I'm trying to store a list of strings (that happen to be UUID's) in a Postgres database with prisma. I really just want to treat them like strings in this case. The problem is when I try and update the list I get:

```
Argument set: Got invalid value 
       |         [
       |           'Some String',
       |           '09F9EEFD-6BF5-46BA-A2C3-703953E03009',
       |           '1FB46A09-2D9B-4471-A3ED-07D82EC51E77'
       |         ]
       |         on prisma.updateOneUser. Provided List<UUID | String>, expected List<String>.
```

I'm not sure how to tell prisma to just treat these as strings?

**Comment by ryands17** [comment]
Hey @chrisdrackett :wave: 
Could you try converting them to plain strings before storing them in the DB as the type looks to be UUID.
Something like:
```ts
value.toString()
```

So that it is stored as a pure string and not a UUID.

Let me know if that works for you. If not, I would need your schema and query that you are performing to save these UUID strings to replicate.

**Reply by chrisdrackett** [reply]
I'm attempting to do the following:

```
prisma db.user.update({
          where: { id: userId },
          data: {
            settingCalendarsToSync: { set: settingCalendarsToSync },
          },
        })
```

and in this case `settingCalendarsToSync` is an array of strings. My guess is that somehow prisma is seeing that some of the strings look like UUID's and then throws the error.

**Reply by ryands17** [reply]
@chrisdrackett Below is the schema I have created to test this:

```prisma
model User {
  id     Int      @default(autoincrement()) @id
  values String[]
}
```

And this is the query that I have run after adding random UUID's:

```ts
import * as Prisma from '@prisma/client'
import { v4 } from 'uuid'

const prisma = new Prisma.PrismaClient({
  log: ['query'],
})

async function main() {
  let items = []
  for (let i of [1, 2, 3, 4, 5]) {
    items.push(v4())
  }
  await prisma.user.create({
    data: {
      values: {
        set: items,
      },
    },
  })
}

main()
  .catch((e) => console.error('e', e))
  .finally(async () => await prisma.disconnect())
```

This doesn't throw any error regarding UUID's. And I am on version 2.0.0 of `@prisma/cli` and `@prisma/client`.

Do let me know if this is correct.

**Reply by chrisdrackett** [reply]
does it still work if you put a non-UUID in that column? This is where I'm running into issues. If I remove all strings or all UUID-looking strings from the database it works as expected. Its the mix that causes the issue

**Reply by ryands17** [reply, marked as answer]
@chrisdrackett I can reproduce this. Which is why I have created an issue here: https://github.com/prisma/prisma/issues/2784

**Comment by ludralph** [comment]
Hi @chrisdrackett,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
